usr/share/rear/lib/global-functions.sh

```diff
@@ -821,37 +821,71 @@ function umount_davfs() {
         fi
     fi
 
-    ### First, try a normal unmount,
+    ### First, try a normal unmount using a timeout in case mountpoint became unresponsive
+    ### due to QoS squeezing or due to stale NFS as the NFS server became unreachable (during mkbackup)
+    ### That is the reason why we use a timeout in front of the umount command.
+    ### However, when tar is busy and the NFS becomes stale then ReaR processes will just hang forever
+    ### until we kill them manually.
     Log "Unmounting '$mountpoint'"
-    umount $v "$mountpoint" >&2
+    timeout $timeout_secs umount $v "$mountpoint" >&2
     if [[ $? -eq 0 ]] ; then
         return 0
     fi
 
-    ### otherwise, try to kill all processes that opened files on the mount.
-    # TODO: actually implement this
-
     ### If that still fails, force unmount.
-    Log "Forced unmount of '$mountpoint'"
-    umount $v -f "$mountpoint" >&2
+    Log "Forcing unmount of '$mountpoint'"
+    timeout $timeout_secs umount $v --force "$mountpoint" >&2
     if [[ $? -eq 0 ]] ; then
         return 0
     fi
 
-    Log "Unmounting '$mountpoint' failed."
+    Log "Unmounting '$mountpoint' failed (after force umount)."
 
-    if test $lazy ; then
-        umount_mountpoint_lazy "$mountpoint"
-    else
-        return 1
+    Log "$mountpoint is still in use by ('kernel mount' is always there)"
+    # The -M option avoids that fuser may show all processes using the '/' filesystem
+    # e.g. for mountpoint $TMP_DIR/somedir ($TMP_DIR = $BUILD_DIR/tmp = /var/tmp/rear.XXXXXXXXXXXXXXX/tmp/)
+    # when $TMP_DIR/somedir got umounted just before fuser starts, see "man fuser":
+    #   The mount -m option will match any file within the same device as the specified file,
+    #   use the -M option as well if you mean to specify only the mount point.
+    # So when $TMP_DIR/somedir is umounted 'fuser -v -M -m $TMP_DIR/somedir' only shows
+    #   "Specified filename /var/tmp/rear.XXXXXXXXXXXXXXX/tmp/somedir is not a mountpoint"
+    # instead of all processes using '/' (or /var/ or /var/tmp/ if one is a mountpoint)
+    # which would be misleading information that may even look scaring and cause false alarm.
+    # Older systems do not support -M but we must use it to avoid misleading information or false alarm.
+    # Since this code path is exceptional and the output is used only for information and only in the log file
+    # we do not care when fuser fails with "M: unknown signal; fuser -l lists signals":
+    fuser -v -M -m "$mountpoint" 1>&2 || Log "Presumably 'fuser' does not support the -M option"
+
+    # Lazy umount only hides the filesystem from new processes.
+    LogPrint "Trying a 'lazy' umount on '$mountpoint' (as it could be stale)."
+    timeout $timeout_secs umount $v --lazy "$mountpoint" >&2
+    if [[ $? -eq 0 ]] ; then
+	return 0
     fi
+
+    Error "Unmounting '$mountpoint' failed even after a force and lazy umount."
```

Comment:
This should be only a LogPrintError
with 'return 1' to let the caller of the function
decide what to do when umounting failed.

Normally there should be no error exit inside a function
but a function should return non-zero exit code
and leave it to its caller what to do depending
on the caller's environment.

In general I am against to error out in any case
only because some 'umount' was not possible
unless there is a reason that a particular 'umount'
must be successful to let ReaR proceed.

Normally nothing should go wrong when something
is still mounted until ReaR exist because
when ReaR exits then all its own processes
(i.e. all child processes of ReaR's MASTER_PID)
get terminated so there is normally no reason to
error out because some 'umount' was not possible.